### PR TITLE
Bonds to R groups should be generic

### PR DESCRIPTION
**Pull request by ptosco**
`generateDepictionMatching2DStructure` has an `allowOptionalAttachments` parameter which allows using scaffold bearing R groups. However, in most cases the bonds connecting R groups to the scaffold are single bonds, which won't match aromatic bonds. This is undesirable as the goal is to make depictions as consistent as possible in a SAR table to facilitate reading.
This PR corrects this by allowing single bonds from aromatic rings to R groups to match single or aromatic bonds using the existing `AdjustQueryParameters`.

### CODE REVIEW

File: Code/GraphMol/Depictor/RDDepictor.cpp [hunk 1]
```diff
@@ -815,12 +815,17 @@ RDKit::MatchVectType generateDepictionMatching2DStructure(
   if (allowOptionalAttachments) {
     std::unique_ptr<RDKit::ROMol> molHs(RDKit::MolOps::addHs(mol));
     CHECK_INVARIANT(molHs, "addHs returned a nullptr");
-    auto matches = SubstructMatch(*molHs, query);
+    auto queryParams = RDKit::MolOps::AdjustQueryParameters::noAdjustments();
+    queryParams.makeBondsGenericFlags = RDKit::MolOps::ADJUST_IGNORENONDUMMIES;
+    queryParams.makeBondsGeneric = true;
```
[the remark this answers is not in the document]
Comment (ptosco) [reply Code/GraphMol/Depictor/RDDepictor.cpp:820]: The way `AdjustQueryProperties` works is that you can choose what you want to adjust, and to what extent, with `AdjustQueryParameters`. So you set `makeBondsGeneric` to `true` (that option pre-existed my PR), and with `makeBondsGenericFlags` you specify that you only want to operate on bonds to dummy atoms (`ADJUST_IGNORENONDUMMIES`). This means that only bonds to dummy atoms will become generic.

[the remark this answers is not in the document]
Comment (bp-kelley) [reply Code/GraphMol/Depictor/RDDepictor.cpp:820]: Ah, I missed the fact that makeBondsGeneric already existed.

File: Code/GraphMol/Substruct/SubstructUtils.cpp [hunk 2]
```diff
@@ -42,13 +42,12 @@ class ScoreMatchesByDegreeOfCoreSubstitution {
         d_minIdx(-1),
         d_isSorted(false) {
     PRECONDITION(!matches.empty(), "matches must not be empty");
-    for (unsigned int i = 0; i < d_mol.getNumAtoms(); ++i) {
-      d_sumIndices += static_cast<double>(i);
-    }
+    auto na = d_mol.getNumAtoms();
+    d_sumIndices = static_cast<double>(na * (na + 1) / 2);
```
Comment (greglandrum) [inline Code/GraphMol/Substruct/SubstructUtils.cpp:46]: :facepalm:

File: Code/MinimalLib/common.h [hunk 3]
```diff
@@ -890,10 +891,17 @@ std::string generate_aligned_coords(ROMol &mol, const ROMol &templateMol,
     if (allowRGroups) {
       molHs.reset(MolOps::addHs(mol));
       prbMol = molHs.get();
+      auto queryParams = RDKit::MolOps::AdjustQueryParameters::noAdjustments();
```
Comment (greglandrum) [inline Code/MinimalLib/common.h:894]: To think about for the future: it's worth figuring out whether or not we can get rid of some of the duplicate chemical logic that's present in this file

**Review by greglandrum — APPROVED** [review]
LGTM

**Comment by greglandrum** [comment]
> @greglandrum Upon reflection, this could be implemented without adding any new flags by using the existing `AdjustQueryParameters`. Simpler and cleaner.

indeed. I'm a lot more comfortable with it this way!

File: Code/MinimalLib/common.h (hunk 3 above)
Comment (ptosco) [reply Code/MinimalLib/common.h:894]: Yeah, I have had this in the back of my mind for a while now, your comment was the final push to tip it over the activation barrier.
I have just submitted #6368 to fix that.

**Pull request merged**